forge/test/operators/pytorch/eltwise_nary/test_concatenate.py

```diff
@@ -0,0 +1,311 @@
+
+
+class ModelDirect(torch.nn.Module):
+
+    model_name = "model_op_src_from_host"
+
+    def __init__(self, operator, opname, shape, number_of_operands, kwargs):
+        super(ModelDirect, self).__init__()
+        self.testname = "Concatenate_pytorch_operator_" + opname + "_test_op_src_from_host"
+        self.operator = operator
+        self.opname = opname
+        self.shape = shape
+        self.number_of_operands = number_of_operands
+        self.kwargs = kwargs
+
+    def forward(self, *x: torch.Tensor):
+        output = self.operator(x, **self.kwargs)
+        return output
+
+
+class ModelConstEvalPass(torch.nn.Module):
+
+    model_name = "model_op_src_const_eval_pass"
+
+    def __init__(self, operator, opname, shape, number_of_operands, kwargs):
+        super(ModelConstEvalPass, self).__init__()
+        self.testname = "Concatenate_pytorch_operator_" + opname + "_test_op_src_const_eval_pass"
+        self.operator = operator
+        self.opname = opname
+        self.shape = shape
+        self.constant_shape = ShapeUtils.reduce_microbatch_size(shape)
+        self.kwargs = kwargs
+
+        self.constants = tuple()
+        for _ in range(number_of_operands):
+            self.constants += (torch.rand(*self.constant_shape) - 0.5,)
+
+    def forward(self, *x: torch.Tensor):
+        v1 = self.operator(self.constants, **self.kwargs)
+        v2 = torch.add(x[0], x[1])
+        for i in range(2, len(x)):
+            v3 = torch.add(v2, x[i])
+            v2 = v3
+        # add consume inputs
+        add = torch.add(v1, v2)
+        return add
+
+
+class TestVerification:
+
+    MODEL_TYPES = {
+        InputSource.FROM_ANOTHER_OP: ModelFromAnotherOp,
+        InputSource.FROM_HOST: ModelDirect,
+        InputSource.FROM_DRAM_QUEUE: ModelDirect,
+        InputSource.CONST_EVAL_PASS: ModelConstEvalPass,
+    }
+
+    @classmethod
+    def verify(
+        cls,
+        test_device: TestDevice,
+        test_vector: TestVector,
+        input_params: List[Dict] = [],
+        warm_reset: bool = False,
+    ):
+        """Common verification function for all tests"""
+
+        input_source_flag: InputSourceFlags = None
+        if test_vector.input_source in (InputSource.FROM_DRAM_QUEUE,):
+            input_source_flag = InputSourceFlags.FROM_DRAM
+
+        operator = getattr(torch, test_vector.operator)
+
+        kwargs = test_vector.kwargs if test_vector.kwargs else {}
+
+        model_type = cls.MODEL_TYPES[test_vector.input_source]
+        pytorch_model = model_type(
+            operator=operator,
+            opname=test_vector.operator,
+            shape=test_vector.input_shape,
+            number_of_operands=test_vector.number_of_operands,
+            kwargs=kwargs,
+        )
+
+        dim = test_vector.kwargs["dim"]
+        match test_vector.input_source:
+            case InputSource.CONST_EVAL_PASS:
+                input_shape = list(ShapeUtils.reduce_microbatch_size(test_vector.input_shape))
```

Comment:
Reducing microbatch size for inputs effectively tests only microbatch size = 1 scenario